cmd/k8s-node-label/main.go

```diff
@@ -28,11 +44,83 @@ func main() {
 		log.SetLevel(log.InfoLevel)
 	}
 
+	if len(*leaseLockNamespace) == 0 {
+		log.Fatal("Flag lease-lock-namespace is not set and default value is not available")
+		os.Exit(1)
+	}
+
 	client, err := common.ClientSet(*kubeconfig)
 	if err != nil {
 		log.Fatalf("Failed to create Kubernetes client %v", err)
+		os.Exit(1)
+	}
+
+	spotProvider, err := spotdiscovery.SpotProviderFactory(*provider)
+	if err != nil {
+		log.Fatalf("can't get spot provider client: %v", err)
+		os.Exit(1)
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	leaseLock := &resourcelock.LeaseLock{
+		LeaseMeta: metav1.ObjectMeta{
+			Name:      *leaseLockName,
+			Namespace: *leaseLockNamespace,
+		},
+		Client: client.CoordinationV1(),
+		LockConfig: resourcelock.ResourceLockConfig{
+			Identity: *leaseId,
+		},
 	}
 
-	spotProvider := spotdiscovery.SpotProviderFactory(*provider)
-	controller.NewNodeController(client, spotProvider, *excludeNodeFromLoadbalancer, *alphaFlags, *excludeEviction, *controlPlaneTaint, *controlPlaneLegacyLabel, *customRoleLabel).Controller.Run(wait.NeverStop)
+	// start the leader election code loop
+	leaderelection.RunOrDie(ctx, leaderelection.LeaderElectionConfig{
+		Lock: leaseLock,
+		// IMPORTANT: you MUST ensure that any code you have that
+		// is protected by the lease must terminate **before**
+		// you call cancel. Otherwise, you could have a background
+		// loop still running and another process could
+		// get elected before your background loop finished, violating
+		// the stated goal of the lease.
+		ReleaseOnCancel: true,
+		LeaseDuration:   60 * time.Second,
+		RenewDeadline:   15 * time.Second,
+		RetryPeriod:     5 * time.Second,
+		Callbacks: leaderelection.LeaderCallbacks{
+			OnStartedLeading: func(ctx context.Context) {
+				// we're notified when we start - this is where you would
+				// usually put your code
+				// run(ctx)
```

Comment:
Can we please add some logging here as well

Reply:
Implemented